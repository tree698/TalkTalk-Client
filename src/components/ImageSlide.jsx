import React, { useEffect } from 'react';
import { useQuery } from '@tanstack/react-query';
import Carousel from 'react-material-ui-carousel';
import toast from 'react-hot-toast';
import { SlUserFollowing } from 'react-icons/sl';
import { MdDownloading } from 'react-icons/md';
import { BiTime } from 'react-icons/bi';
import { formatAgo } from '../util/date';
import { useApiContext } from '../context/ApiContext';
import { paginationForCarousel } from '../config';
import Banner from './ui/Banner';

export default function ImageSlide() {
  const { workService } = useApiContext();
  const { limit, offset } = paginationForCarousel;
  const { isLoading, error, data: images } = useQuery(
    ['carousel'],
    async () => await workService.showWorks(limit, offset)
  );

  useEffect(() => {
    error && toast.error(error.message);
  }, [error]);

  return (
    <section className="flex justify-center">
      <Carousel
        autoPlay={true}
        stopAutoPlayOnHover={true}
        indicators={false}
        animation="fade"
        navButtonsAlwaysVisible={false}
        className="w-[320px] md:w-[420px] lg:w-[520px] h-[240px] md:h-[315px] lg:h-[390px]"
      >
        {images &&
          images.map((image) => (
            <div
              key={image.id}
              className="w-[320px] md:w-[420px] lg:w-[520px] h-[240px] md:h-[315px] lg:h-[390px]"
            >
              <img
                src={image.fileName}
                alt={image.title}
                style={{ width: '100%', height: '100%', borderRadius: '3%' }}
                className="border border-superLightGray shadow-md"
              />
              <div className="flex justify-center text-center text-xs md:text-sm lg:text-lg text-gray-600 mt-1 md:mt-2 lg:mt-3">
                <div className="flex justify-center items-center mr-2 md:mr-4 lg:mr-6">
                  <SlUserFollowing />
                  <span className="ml-1 md:ml-2">{image.username}</span>
                </div>
                <div className="flex justify-center items-center">
                  <BiTime />
                  <span className="ml-1 md:ml-2">
                    {formatAgo(image.createdAt)}
                  </span>
                </div>
              </div>
            </div>
          ))}
      </Carousel>
      {isLoading && (
        <div className="flex items-center justify-center gap-4 mt-4 md:mt-6 lg:mt-8 text-xs md:text-sm lg:text-lg">
          <MdDownloading className="text-lg md:text-base lg:text-xl" />
          <Banner text="Loading..." />
        </div>
      )}
    </section>
  );
}
