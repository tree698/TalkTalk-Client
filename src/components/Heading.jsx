import React from 'react';
import ImageSlide from './ImageSlide';

export default function Heading() {
  return (
    <section className="w-full basis-1/2 bg-cover bg-headingBG lg:basis-3/5 flex flex-col justify-center p-4 pt-10">
      <img
        src="images/logo.png"
        alt="logo"
        className="w-1/2 mx-auto md:w-[360px] lg:w-[400px]"
      />
      <h2 className="text-base text-center md:text-2xl lg:text-3xl my-3 md:my-6 lg:my-9">
        Upload drawing, then enjoy talking
      </h2>
      <ImageSlide />
    </section>
  );
}
