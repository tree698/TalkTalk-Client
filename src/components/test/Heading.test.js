import ImageSlide from '../ImageSlide';
import renderer from 'react-test-renderer';
import { withRouter } from '../../test/utils';
import { Route } from 'react-router-dom';
import Heading from '../Heading';

jest.mock('../ImageSlide');

describe('Heading', () => {
  ImageSlide.mockImplementation(() => jest.fn());

  it('renders correctly', () => {
    const component = renderer.create(
      withRouter(<Route path="/" element={<Heading />} />)
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
