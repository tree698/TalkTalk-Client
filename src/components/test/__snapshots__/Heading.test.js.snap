// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders correctly 1`] = `
<section
  className="basis-1/2 lg:basis-3/5 w-full flex flex-col justify-center items-center bg-cover bg-headingBG  md:mx-2 lg:mx-0"
>
  <img
    alt="logo"
    className="w-[30%] md:w-[40%]"
    src="images/logo.png"
  />
  <h2
    className="text-xs md:text-sm lg:text-base mb-4 md:mb-6 lg:mb-8"
  >
    Upload drawing, then enjoy talking
  </h2>
</section>
`;
